cubi_tk/snappy/common.py

```diff
@@ -37,6 +23,124 @@ class CouldNotFindBioMedSheet(Exception):
     """Raised when BioMedSheet could not be found in configuration file."""
 
 
+class DummyWorkflow:
+    """Dummy workflow that does nothing."""
+
+    def __init__(self):
+        self.globals = defaultdict(str)
+
+    def __getattr__(self, _):
+        return self._catchall
+
+    def _catchall(self, *_, **__):
+        pass
+
+
+def get_available_snappy_workflows():
+    """Get all available snappy workflows in snappy_pipeline.
+
+    :return: List of ModuleInfo with individual workflow modules.
+    """
+    return list(pkgutil.iter_modules(snappy_workflows.__path__))
+
+
+def get_available_snappy_workflow_paths():
+    """Get module paths for all available snappy workflows.
+
+    :return: Dict of workflow name to workflow module path.
+    """
+    workflow_names = get_available_snappy_workflows()
+    return {w.name: pathlib.Path(w.module_finder.path) / w.name for w in workflow_names}
+
+
+def get_workflow_snakefile_object_name(snakefile_path):
+    """Find the Workflow implementation object name.
+
+    :param snakefile_path: Path to snakefile for workflow.
+    :type snakefile_path: str, pathlib.Path
+
+    :return: str Name of the implementation class or None if nothing as been found.
+    """
+
+    with snakefile_path.open() as f:
+        if m := re.search(r"wf\s*=\s*(\w+)\(", f.read()):
+            module_name = m.group(1)
+            return module_name
+    return None
+
+
+def load_workflow_step_configuration(workflow_step_path):
+    """Load snappy configuration and resolve any refs.
+
+    :param workflow_step_path: Path to snappy config yaml.
+    :type workflow_step_path: str, pathlib.Path
+
+    :return: Tuple of config, lookup paths and configuration paths. If no config is found, a tuple of None is returned.
+    """
+
+    config_path = pathlib.Path(workflow_step_path) / "config.yaml"
+
+    if not config_path.exists():
+        return (None, None, None)
+
+    with open(str(config_path)) as stream:
+        config_data = yaml.safe_load(stream)
+
+    config, lookup_paths, config_paths = expand_ref(
+        str(config_path),
+        config_data,
+        [str(workflow_step_path), str(workflow_step_path.parent / ".snappy_pipeline")],
+    )
+    return config, lookup_paths, config_paths
```

Comment:
The `config` object returned as first element of the retruned tuple here is a string. However, most functions working with this expect it to be a (yaml representing) dictionary, likely objects are switched?